File: examples/key/encrypt_rsa/encrypt_rsa.ts
const {
  EncryptionClient,
  RecordClient,
  KeyClient,
  RsaEncrypter,
} = require('@bloock/sdk');

let payload = 'This will be encrypted';

console.log(`The following payload will be encrypted: ${payload}`);

const encryptionClient = new EncryptionClient();
const keyClient = new KeyClient();
const recordClient = new RecordClient();

let key = await keyClient.newLocalKey(KeyType.Rsa2048);

// To encrypt a record during the building process
let encryptedRecord = await recordClient
  .fromString(payload)
  .withEncrypter(new RsaEncrypter(key))
  .build();

// To encrypt a record independently
let record = await recordClient.fromString(payload).build();
encryptedRecord = await encryptionClient.encrypt(record, new RsaEncrypter(key));

console.log(`Encryption successfully`);

let encryptedPayload = encryptedRecord.retrieve();
console.log(`Encrypted payload: ${new TextDecoder().decode(encryptedPayload)}`);

File: examples/key/encrypt_rsa/encrypt_rsa.go
package main

import (
	"fmt"
	"log"

	"github.com/bloock/bloock-sdk-go/v2/client"
	"github.com/bloock/bloock-sdk-go/v2/entity/encryption"
	"github.com/bloock/bloock-sdk-go/v2/entity/key"
)

func main() {
	payload := "This will be encrypted"
	encryptionClient := client.NewEncryptionClient()
	recordClient := client.NewRecordClient()
	keyClient := client.NewKeyClient()

	key, err := keyClient.NewLocalKey(key.Rsa2048)
	if err != nil {
		log.Println(err)
	}

	fmt.Printf("The following payload will be encrypted: %s", payload)

	// To encrypt a record during the building process
	encryptedRecord, err := recordClient.FromString(payload).
		WithEncrypter(encryption.NewEncrypterWithLocalKey(key)).
		Build()
	if err != nil {
		log.Println(err)
	}

	// To encrypt a record independently
	record, err := recordClient.FromString(payload).Build()
	if err != nil {
		log.Println(err)
	}
	encryptedRecord, err = encryptionClient.Encrypt(record, encryption.NewEncrypterWithLocalKey(key))
	if err != nil {
		log.Println(err)
	}

	fmt.Println("Encryption successful")
	fmt.Println("Encrypted payload: " + string(encryptedRecord.Retrieve()))
}
